cosmwasm/lib/Makefile

```diff
@@ -0,0 +1,172 @@
+ifeq ($(SGX_PRERELEASE), 1)
+$(error Cannot set SGX_DEBUG and SGX_PRERELEASE at the same time!!)
+endif
+endif
+
+
+ifeq ($(SGX_DEBUG), 1)
+	SGX_COMMON_CFLAGS += -O0 -g
+else
+	SGX_COMMON_CFLAGS += -O2
+endif
+
+######## CUSTOM Settings ########
+
+CUSTOM_LIBRARY_PATH := ./lib
+CUSTOM_BIN_PATH := ./bin
+CUSTOM_EDL_PATH := ./incubator-teaclave-sgx-sdk/edl
+CUSTOM_COMMON_PATH := ./incubator-teaclave-sgx-sdk/common
+
+######## EDL Settings ########
+
+Enclave_EDL_Files := ./wasmi-runtime/Enclave_t.c ./wasmi-runtime/Enclave_t.h ./sgx-vm/Enclave_u.c ./sgx-vm/Enclave_u.h
+
+######## APP Settings ########
+
+App_Rust_Flags := --release
+App_SRC_Files := $(shell find sgx-vm/ -type f -name '*.rs') $(shell find sgx-vm/ -type f -name 'Cargo.toml')
+App_Include_Paths := -I ./sgx-vm -I./include -I$(SGX_SDK)/include -I$(CUSTOM_EDL_PATH)
+App_C_Flags := $(SGX_COMMON_CFLAGS) -fPIC -Wno-attributes $(App_Include_Paths)
+
+App_Rust_Path := ./sgx-vm/target/release
+App_Enclave_u_Object :=sgx-vm/libEnclave_u.a
+App_Name := bin/app
+
+######## Enclave Settings ########
+
+ifneq ($(SGX_MODE), HW)
+	Trts_Library_Name := sgx_trts_sim
+	Service_Library_Name := sgx_tservice_sim
+else
+	Trts_Library_Name := sgx_trts
+	Service_Library_Name := sgx_tservice
+endif
+Crypto_Library_Name := sgx_tcrypto
+KeyExchange_Library_Name := sgx_tkey_exchange
+ProtectedFs_Library_Name := sgx_tprotected_fs
+
+RustEnclave_C_Files := $(wildcard ./wasmi-runtime/*.c)
+RustEnclave_C_Objects := $(RustEnclave_C_Files:.c=.o)
+RustEnclave_Include_Paths := -I$(CUSTOM_COMMON_PATH)/inc -I$(CUSTOM_EDL_PATH) -I$(SGX_SDK)/include -I$(SGX_SDK)/include/tlibc -I$(SGX_SDK)/include/stlport -I$(SGX_SDK)/include/epid -I ./wasmi-runtime -I./include
+
+RustEnclave_Link_Libs := -L$(CUSTOM_LIBRARY_PATH) -lcompiler-rt-patch -lenclave
+RustEnclave_Compile_Flags := $(SGX_COMMON_CFLAGS) -nostdinc -fvisibility=hidden -fpie -fstack-protector $(RustEnclave_Include_Paths)
+RustEnclave_Link_Flags := $(SGX_COMMON_CFLAGS) -Wl,--no-undefined -nostdlib -nodefaultlibs -nostartfiles -L$(SGX_LIBRARY_PATH) \
+	-Wl,--whole-archive -l$(Trts_Library_Name) -Wl,--no-whole-archive \
+	-Wl,--start-group -lsgx_tstdc -l$(Service_Library_Name) -l$(Crypto_Library_Name) $(RustEnclave_Link_Libs) -Wl,--end-group \
+	-Wl,-Bstatic -Wl,-Bsymbolic -Wl,--no-undefined \
+	-Wl,-pie,-eenclave_entry -Wl,--export-dynamic  \
+	-Wl,--defsym,__ImageBase=0 \
+	-Wl,--gc-sections \
+	-Wl,--version-script=wasmi-runtime/Enclave.lds
+
+RustEnclave_Name := wasmi-runtime/enclave.so
+Signed_RustEnclave_Name := bin/enclave.signed.so
+
+.PHONY: all
+all: enclave-ffi-types $(App_Name) $(Signed_RustEnclave_Name)
+
+.PHONY: enclave-ffi-types
+enclave-ffi-types:
+	cd enclave-ffi-types && cargo build
+	mkdir -p ./sgx-vm/target/headers
+	mkdir -p ./wasmi-runtime/target/headers
+	cp -f ./enclave-ffi-types/target/headers/enclave-ffi-types.h ./wasmi-runtime/target/headers
+	cp -f ./enclave-ffi-types/target/headers/enclave-ffi-types.h ./sgx-vm/target/headers
```

Comment:
All of this already happens as part of `enclave-ffi-types`'s build script, and you don't need to do it at all :) Any package that depends on `enclave-ffi-types` will have these files available when building, before it gets built itself

Reply:
Both packages depend on `enclave-ffi-types` and they don't have these files, so...

Reply:
It doesn't matter

Reply:
resolved in call